### ISSUE TITLE
Regression in jsonpath matching

### ISSUE THREAD

**Issue by marcingrzejszczak** [tomakehurst/wiremock#1492]
# Which version of WireMock you're using

works with 2.27.1
doesn't work with 2.28.0

# Reproducing

Create such a json mapping and upload it to WireMock

```json
{
  "id" : "eef52385-481f-4fb3-813d-ee657b437e9d",
  "request" : {
    "url" : "/api/user",
    "method" : "POST",
    "headers" : {
      "Content-Type" : {
        "matches" : "application/json.*"
      }
    },
    "bodyPatterns" : [ {
      "matchesJsonPath" : "$[?(@.['address'] =~ /([a-zA-Z0-9]+)?/)]"
    }, {
      "matchesJsonPath" : "$[?(@.['name'] =~ /([a-zA-Z0-9]+)?/)]"
    } ]
  },
  "response" : {
    "status" : 201,
  },
  "uuid" : "eef52385-481f-4fb3-813d-ee657b437e9d"
}
```

Send such a POST request to `/api/user`. `Content-Type` is `application/json` and the body 

```json
{ "foo" : null, "name": "" }
```

Attaching a project that reproduces the issue. Just run `./mvnw verify` - the build will fail. Run `./mvnw verify -Dwiremock-jre8-standalone.version=2.27.1` and the build will pass.

**Comment by marcingrzejszczak** [comment]
[jsonpath_regression.zip](https://github.com/tomakehurst/wiremock/files/6472295/jsonpath_regression.zip)

**Comment by tomakehurst** [comment]
This looks like it's been caused by a change to the JSONPath parser in the 2.4.0 -> 2.5.0 upgrade. The JSONPath expression isn't being parsed any more and an exception is being thrown (which WireMock only reports if you've got verbose logging turned on).

The previous behaviour looks wrong to me though, for a couple of reasons:

1) If you're requiring an `address` field to be present and not sending it, that ought to be a non-match.
2) Your `name` regex expects there to be at least one character, so sending an empty name field also ought to cause a non-match.

Removing the capture group brackets from the regex and sending a non-empty name and address makes things work again, so:

```json
{
  "request" : {
    "url" : "/api/user",
    "method" : "POST",
    "headers" : {
      "Content-Type" : {
        "matches" : "application/json.*"
      }
    },
    "bodyPatterns" : [ {
      "matchesJsonPath" : "$[?(@.['address'] =~ /[a-zA-Z0-9]+?/)]"
    }, {
      "matchesJsonPath" : "$[?(@.['name'] =~ /[a-zA-Z0-9]+?/)]"
    } ]
  },
  "response" : {
    "status" : 201
  }
}
```

Will match

```json
{ "foo" : null, "name": "a", "address": "b" }
```

**Comment by tomakehurst** [comment]
Related json-path issue: https://github.com/json-path/JsonPath/issues/708

**Comment by tomakehurst** [comment]
And also https://github.com/json-path/JsonPath/issues/692

**Comment by tomakehurst** [comment]
Fixed in bbd7b2489d515f12ffd69beee3eda78c0bbe3b05
